src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -395,6 +395,16 @@ private void StartAsynchronousComputation(AsynchronousComputationToStart computa
                         // to the caller of GetValueAsync yet
                         requestToCompleteSynchronously.CompleteFromTaskSynchronously(task);
                     }
+
+                    // We avoid creating a full closure just to pass the token along
```

Comment:
It looks right to me.  It's referring to the lambda we pass into the task.ContinueWith.  And that lambda should not cause a closure to be created.